src/Microsoft.AspNetCore.Routing.Abstractions/Endpoint.cs

```diff
@@ -20,5 +29,7 @@ public abstract class Endpoint
         public string DisplayName { get; }
 
         public EndpointMetadataCollection Metadata { get; }
+
+        public override string ToString() => DisplayName ?? base.ToString();
```

Comment:
It didn't make a lot of sense to me that we weren't doing this already. This should make xUnit better at least